bug fixes - cybermod + DoL event + weekly reports and SA effects + RA hormones for females rule applying incorrectly to herms + more (ready for merge)

fcpmanon opened the pull request.

freecitiesdev commented:
The conflict in slaveSummary.tw is pretty massive. Any idea what's going on there?

fcpmanon commented:
Oops, looks like end-of-line conversion problem.  I'll update this branch to not conflict with the latest commits and also correct the line endings.

fcpmanon commented:
updated change list:

assignWidgets:  slaves removed from confinement or schoolroom are assigned to rest;  backwardsCompatibility:  make spire sectors visible if spire upgrade was already purchased (only for saved games from before 0.10.1);  brothelReport:  rivalry > relationship > relation for special treatment - remove $brothelSpots variable due to reordering of weekly reports - SA chooses own clothes - general cleanup;  buildingWidgets:  reduce stretching of arcology table on wide screens - reveal spire sectors if not already shown after spire upgrade (fix for pre-0.10.1 saved games that were previously brought to 0.10.2 using the backwards compatibility update rather than NGP);  clinic:  corrected location of blood treatment idle hint;  clinicReport:  corrected condition for alternative health bonus (nurse muscles > 5 or devotion >= 60, not >= -20) - rivalry > relationship > relation for special treatment - added "assignment has defaulted to rest" text for individual slaves - SA chooses own clothes - general cleanup;  clubReport:  moved validity checks to the top as in other reports - remove $clubSpots variable due to reordering of weekly reports - SA chooses own clothes - general cleanup;  dairyReport:  rivalry > relationship > relation - expand restraints clothing removal - remove $dairySpots due to reordering of weekly reports - SA chooses own clothes - general cleanup;  descriptionWidgets:  fix degradation forehead tattoo text;  dispensary:  return link to Manage Penthouse, not to Main;  fullReport:  SA chooses own clothes;  headGirlSuite:  find and display correct slave name (not activeSlave);  longSlaveDescription:  don't show genome damage (.chem) below 15;  main:  check for bugged duplicate slave IDs (slave object duplication) and display warning if found, for easier debugging;  mainStyleSheet:  eliminate 10% empty margins;  masterSuiteReport:  SA chooses own clothes;  miscWidgets:  typo;  raWidgets:  fix female hormone rule overriding shemale hormone rule - fix "no contraceptives" rule skipping slaves already on contraceptives - show hint text for custom activation type; rulesAssistant:  restore FS condition for toga - correct span placement for clothes rule - add missing closing if for no-rules-defined case;  saChoosesOwnClothes:  moved devotion changes into the passage - incorporate SlavePronouns;  saChoosesOwnJob:  revise text, reinstate devotion/trust bonus;  saGetMilked:  remove $dairySpots variable due to reordering of weekly reports;  saLiveWithHG:  apply diet, long term effects and drug effects - update HG impregnation check to use canImpreg widget;  saRelationships:  check for bugged (half-formed) relationship and display warning if found, for easier debugging;  saServant:  remove $servantsQuarterSpots variable due to weekly report reordering - incorporate SlavePronouns;  saServeThePublic:  remove $clubSpots variable due to weekly report reordering;  saStayConfined:  display "assignment defaulted to rest" text and update needed variables when included by cellblock report - incorporate SlavePronouns;  saTakeClasses:  incorporate SlavePronouns;  saWhore:  remove $brothelSpots variable due to weekly report reordering - reword AE FS text - general cleanup;  schoolroomReport:  rivalry > relationship > relation - display "assignment defaulted to rest" text - SA chooses own clothes - general cleanup;  servantsQuarters:  typo;  servantsQuartersReport:  rivalry > relationship > relation - use canImpreg widget - SA chooses own clothes - general cleanup;  slaveInteract:  allow sending eligible slaves to live with your agent from any existing assignment, instead of only from the penthouse;  slaveSummary:  SA chooses own clothes - revise choosesOwnAssignment display - consolidate Aph and Add abbreviations using color coding (cyan Aph/Aph++ = addicted + applied; Add alone = addicted, not applied);  spaReport:  remove cure line breaks - display "assignment defaulted to rest" text - SA chooses own clothes - general cleanup;  surgeryDegradation - typos
